---
title: "Fig1.G code"
author:
  - name: "Gabbut and Duran-Ferrer, 2025"
    # affiliation: "Institut d'Investigacions Biom√®diques Augusti Pi i Sunyer (IDIBAPS), Barcelona, Spain."
    # email: "maduran@clinic.cat"
date: "30/01/2025"
output: 
    BiocStyle::html_document:
      toc: true
      toc_float: true
      number_sections: true
      code_folding: show
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load data and R libraries

First, we load necessary R libraries.

```{r}
library(data.table)
library(ggpubr)
library(ggbeeswarm)
library(rstatix)
library(janitor)
library(ggforce)
library(openxlsx)
library(dplyr)


options(stringsAsFactors = F,max.print = 10000,error=NULL,
        "openxlsx.dateFormat" = "dd/mm/yyyy"
        )

```

Then, we load the data source necessary to generate all the figures.

```{r}
sheet.names <- getSheetNames("../../Revision/Results/CLL_fCpGs_Gene_expression_bulk/Data_source_gene_expression.xlsx")
sheet.names <- structure(sheet.names,names=sheet.names)
sheet.names

data.source <- lapply(sheet.names, function(sheet.i){
  res <- read.xlsx(xlsxFile = "../../Revision/Results/CLL_fCpGs_Gene_expression_bulk/Data_source_gene_expression.xlsx",
                   sheet = sheet.i,detectDates = T)
})

lapply(data.source,head)

```

# Plotting!

## PP plot of all cases

We can see how genes associated with fCpGs show a lower gene expression than those non-fCpGs genes.(The Illumina array annotation was used to link CpGs to genes).

```{r}
df.pp.plot <- data.source$pp_plot

pp.plot <- 
  df.pp.plot %>%
  ggplot(aes(x=Non.fCpG.genes,
             y=fCpG.genes
  )
  )+
  xlab("Non-fCpG genes")+
  ylab("fCpG genes")+
  geom_abline(slope = 1,
              color="grey80",
              lty="dashed",
              lwd=0.3
  )+
  ggrastr::rasterise(geom_point(size=1.5),dpi=300)+
  # ggtitle(paste0("pp plot (%) of tpms, samples:",length(unique(df.tpms$sample))))+
  theme_bw()+
  theme(text=element_text(size = 5),
        line = element_line(linewidth = 0.3),
        panel.background = element_rect(fill = NA,colour = NA,linewidth = 0.3)
  )
print(pp.plot)
```

## Example of 1 CLL patient

We selected a random patient and plot the gene expression of genes associated with fCpGs and those that are not associated with them. (The Illumina array annotation was used to link CpGs to genes).

```{r}
random.sample.anonymous <- data.source$sample_genes_fCpGs_non.fCpgs$sample



plot.genes.sample <- 
  data.source$sample_genes_fCpGs_non.fCpgs %>%
  ggplot(aes(x = fCpG,
             y = tpm +1
  )
  )+
  geom_boxplot(
    aes(fill=fCpG),#disease_state,sample_tx
    show.legend = F,
    lwd=0.3
  )+
  geom_pwc(method = "wilcox_test",
           size = 0.3,
           p.adjust.method = "none",
           label = "P={p}",
           y.position = log10(8500),
           label.size = 1
  )+
  scale_fill_manual(values = c("orange2","grey70"))+
  coord_cartesian(ylim = c(0,max(data.source$sample_genes_fCpGs_non.fCpgs$tpm))+1)+
  scale_y_log10()+
  annotation_logticks(sides = "l",
                      size = 0.3)+
  ggtitle(paste0("Anonymous=",random.sample.anonymous))+
  theme_bw()+
  theme(legend.position = "none",
        plot.margin = unit(c(0,0,0,0),"pt"),
        text = element_text(size = 6),
        line = element_line(linewidth = 0.3),
        panel.border = element_rect(fill = NA,linewidth = 0.3),
        panel.background = element_blank()
  )
plot.genes.sample
```

## Expression vs number of methylated alleles

All genes related to fCpGs are lowly expressed regardless of the number of alleles methylated. A beta binomial mixture model was used to categorize CpGs, and then Illumina array annotation was used to link CpGs to genes.

```{r}
## plot genes segregated by meth allele of the random sample
plot.meth.alleles.sample <- 
  data.source$sample_meth_alleles_expression %>% 
  ggplot(aes(x = meth,
             y = tpm+1
  )
  )+
  geom_beeswarm(aes(fill=meth),
                color="grey10",
                corral = "wrap",
                pch=21,
                stroke=0.1,
                size=1
  )+
  geom_boxplot(aes(fill=meth),
               alpha=0.2,
               outlier.shape = NA,
               lwd=0.3
  )+
  scale_fill_manual(values = c("0"="#2F7FC2","2"="#BC5250","1"="grey80"))+
  geom_pwc(method = "wilcox_test",
           label = "P={p}",
           size=0.3,
           label.size = 1,
           step.increase = 0.05
  )+
  coord_cartesian(ylim = c(0,max(data.source$sample_genes_fCpGs_non.fCpgs$tpm))+1)+
  scale_y_log10()+
  annotation_logticks(sides = "l",
                      size=0.3)+
  ylab(NULL)+
  xlab("# of fCpG methylated alleles")+
  theme_bw()+
  theme(legend.position = "none",
        plot.margin = unit(c(0,0,0,0),"pt"),
        text = element_text(size = 6),
        line = element_line(linewidth = 0.3),
        panel.border = element_rect(fill = NA,linewidth = 0.3),
        panel.background = element_blank()
  )

print(plot.meth.alleles.sample)
```

## All plots together

Plot all panels together

```{r}
cowplot::plot_grid(pp.plot,
                   plot.genes.sample,
                   plot.meth.alleles.sample,
                   nrow = 1,
                   align = "h",
                   rel_widths = c(1,0.5,0.75)
                   )
```

# Session info

```{r}
print(sessionInfo())

```
